use-cases/school-timetabling/src/main/java/org/acme/schooltimetabling/service/TimeTableDemoResource.java

```
@@ -0,0 +1,217 @@
+import org.eclipse.microprofile.openapi.annotations.responses.APIResponse;
+import org.eclipse.microprofile.openapi.annotations.responses.APIResponses;
+import org.eclipse.microprofile.openapi.annotations.tags.Tag;
+
+@Tag(name = "Demo data sets", description = "Timefold-provided demo school timetable data sets.")
+@Path("demo/datasets")
+public class TimeTableDemoResource {
+
+    public enum DemoDataSet {
+        SMALL,
+        LARGE
+    }
+
+    @APIResponses(value = {
+            @APIResponse(responseCode = "200", description = "List of demo data sets represented as IDs.",
+                    content = @Content(mediaType = MediaType.APPLICATION_JSON,
+                            schema = @Schema(implementation = DemoDataSet.class, type = SchemaType.ARRAY))) })
+    @Operation(summary = "List demo data sets.")
+    @GET
+    public DemoDataSet[] list() {
+        return DemoDataSet.values();
+    }
+
+    @APIResponses(value = {
+            @APIResponse(responseCode = "200", description = "Unsolved demo timetable.",
+                    content = @Content(mediaType = MediaType.APPLICATION_JSON,
+                            schema = @Schema(implementation = TimeTable.class))),
+            @APIResponse(responseCode = "404", description = "Demo data set does not exist.",
+                    content = @Content(mediaType = MediaType.TEXT_PLAIN))})
+    @Operation(summary = "Find an unsolved demo timetable by ID.")
+    @GET
+    @Path("/{id}")
+    public Response generate(@Parameter(description = "Unique identifier of the demo data set.",
+            required = true) @PathParam("id") String dataSetId) {
+        DemoDataSet demoDataSet;
+        try {
+            demoDataSet = DemoDataSet.valueOf(dataSetId);
+        } catch (IllegalArgumentException illegalArgumentException) {
+            return Response.status(Response.Status.NOT_FOUND).entity("Demo data set " + dataSetId + " not found.").build();
```

n the quickstarts, I wouldn't bother with Response objects. It just makes it more complex.

Just return TimeTable type and throw an IllegalArgumentException. Let the default exception handlers turn it into a 500 (which is good enough). No need to open API document what happens that case.

=> Less code to grok for users.

---

We should fail fast on an invalid dataSetId, but I'd argue that elaborate response statuses etc are out of scope, unless we also start pulling validation etc into scope - which will make the code harder to read.

---

Reworked to use exception mappers as agreed.